qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -253,8 +253,8 @@ def settings(self) -> dict[str, Any]:
         return settings
 
     @staticmethod
-    @deprecate_arg("backend", since="0.22", pending=True)
-    @deprecate_arg("expectation", since="0.22", pending=True)
+    @deprecate_arg("backend", since="0.24", additional_msg="See http://qisk.it/algo_migration for a migration guide.")
+    @deprecate_arg("expectation", since="0.24", additional_msg="See http://qisk.it/algo_migration for a migration guide.")
```

Comment:
I see here its 0.24 whereas we seem elsewhere to do 0.24.0 (maybe it was just a simple replace 0.22 by 0.24 done) but I guess it would better to be consistent